framework/src/meshgenerators/ElementDeletionGeneratorBase.C

```diff
@@ -52,10 +52,15 @@ ElementDeletionGeneratorBase::generate()
       deleteable_elems.insert(elem);
   }
 
-  /**
-   * If we are in parallel we'd better have a consistent idea of what
-   * should be deleted.  This can't be checked cheaply.
-   */
+  // check for dangling interior parents
+  for (auto & elem : mesh->element_ptr_range())
+    if (elem->dim() == mesh->mesh_dimension() - 1 &&
+        deleteable_elems.count(elem->interior_parent()) > 0)
+      deleteable_elems.insert(elem);
+      /**
+       * If we are in parallel we'd better have a consistent idea of what
+       * should be deleted.  This can't be checked cheaply.
+       */
```

Comment:
```suggestion
  /**
   * If we are in parallel we'd better have a consistent idea of what
   * should be deleted.  This can't be checked cheaply.
   */
```